### PR TITLE
TRI-63_Get_/me

### PR DESCRIPTION
**Pull request by Juan-Bianchi**
# Pull Request

## Description
Add domain directory and put all entities within, create user controller and add service and repository for the endpoint. Add auxiliary UserProjectRole class, add NotFoundException. Add two eslint rules in .eslintrc.json. Create tests.

## Ticket Link
[TRI-63](https://linear.app/tricker-sirius/issue/TRI-63/get-me)

## Steps to Reproduce
1. Run `docker compose build` (config files have been changed) to re build your container.
2. Run  `npm run test`
3. User service tests should pass. 
4. Check  in ./src/domains/user/user.controller.ts    get /me enpoint,
./src/domains/user/user.service.impl.ts    getById method,
./src/domains/user/user.repository.impl.ts    getById method.
5. Run `docker compose up`, and click on this link: [API documentation page](http://localhost:8080/api-docs/#/)

### CODE REVIEW

File: tests/domains/user/service/user.service.impl.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  });
+
+  beforeEach(() => {
+    mock.restoreAll();
+  });
+
+  it('Should successfully get a user by a provided id', async () => {
+    const recievedUser: UserDTO = await service.getById('id');
+
+    assert.notEqual(recievedUser, null);
```
Comment (luccar31) [inline tests/domains/user/service/user.service.impl.test.ts:33]: Redundant validation. If user is null, id won't be equal to the provided one

File: src/domains/user/controller/user.controller.ts
Comment (luccar31) [inline src/domains/user/controller/user.controller.ts:1]: Missing documentation for this endpoint

**Review by luccar31 — CHANGES_REQUESTED** [review]

File: tests/domains/user/service/user.service.impl.test.ts (hunk 1 above)
Comment (Juan-Bianchi) [reply tests/domains/user/service/user.service.impl.test.ts]: Fixed

File: src/domains/user/controller/user.controller.ts
Comment (Juan-Bianchi) [reply src/domains/user/controller/user.controller.ts:1]: fixed

**Review by luccar31 — APPROVED** [review]

**Comment by FedericoMartucci** [comment]
I'm unable to test the endpoints within Swagger. When it throws a 404, it doesn't match the expected response. Verify if the 'deletedAt' property in UserDTO is necessary.

**Comment by Juan-Bianchi** [comment]
Fix swagger functionallity, it is working ok. deletedAt will be used in this DTO in the future.

**Review by FedericoMartucci — APPROVED** [review]

**Pull request merged**